File: event-bus/src/main/java/com/github/dynamicextensionsalfresco/event/impl/Activator.java
package com.github.dynamicextensionsalfresco.event.impl;

import com.github.dynamicextensionsalfresco.event.EventBus;
import org.jetbrains.annotations.NotNull;
import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

/**
 * @author Laurent Van der Linden
 */
public final class Activator implements BundleActivator {

    public void start(@NotNull BundleContext context) {
        if (context == null) {
            throw new IllegalArgumentException("context is null");
        }
        context.registerService(EventBus.class, new DefaultEventBus(context), null);
    }

    public void stop(@NotNull BundleContext context) {
        if (context == null) {
            throw new IllegalArgumentException("context is null");
        }
    }
}

File: event-bus/src/main/kotlin/com/github/dynamicextensionsalfresco/event/impl/Activator.kt
package com.github.dynamicextensionsalfresco.event.impl

import com.github.dynamicextensionsalfresco.event.EventBus
import org.osgi.framework.BundleActivator
import org.osgi.framework.BundleContext

/**
 * @author Laurent Van der Linden
 */
public class Activator : BundleActivator {
    throws(Exception::class)
    override fun start(context: BundleContext) {
        context.registerService(javaClass<EventBus>(), DefaultEventBus(context), null)
    }

    throws(Exception::class)
    override fun stop(context: BundleContext) {
    }
}
